cloudify_agent/tests/test_operations.py

```diff
@@ -70,17 +71,42 @@ def _manager_env(self):
         shutil.copyfile(agent_package.get_package_path(), agent_path)
 
         new_env = {
-            constants.REST_HOST_KEY: 'localhost',
-            constants.MANAGER_FILE_SERVER_URL_KEY:
-                'http://localhost:{0}'.format(port),
             constants.MANAGER_FILE_SERVER_ROOT_KEY: resources_dir,
             constants.REST_PORT_KEY: str(port),
         }
-        with patch.dict(os.environ, new_env):
-            try:
-                yield
-            finally:
-                fs.stop()
+
+        # Necessary to patch, because by default https will be used
+        def file_server_url(*args, **kwargs):
+            return 'http://localhost:{0}/resources'.format(port)
+
+        # Need to patch, to avoid broker_ssl_enabled being True
+        @contextmanager
+        def get_amqp_client(agent):
+            yield get_client()
+
+        managers = [
+            ManagerItem({
+                'networks': {'default': '127.0.0.1'},
+                'ca_cert_content': agent_ssl_cert.DUMMY_CERT
+            })
+        ]
+        patches = [
+            patch.dict(os.environ, new_env),
+            patch('cloudify_agent.api.utils.get_manager_file_server_url',
+                  file_server_url),
+            patch('cloudify_agent.operations._get_amqp_client',
+                  get_amqp_client),
+            patch('cloudify.endpoint.LocalEndpoint.get_managers',
+                  return_value=managers),
+        ]
```

Comment:
added another patch so factored the patches here to avoid the 1000 nested `with`s